proparse/src/main/java/org/prorefactor/proparse/antlr4/Lexer.java

```diff
@@ -0,0 +1,1117 @@
+  }
+
+  ProToken whitespace() throws IOException {
+    LOGGER.trace("Entering whitespace()");
+
+    loop : for (;;) {
+      switch (currChar) {
+        case ' ':
+        case '\t':
+        case '\f':
+        case '\n':
+        case '\r':
+          append();
+          getChar();
+          break;
+        default:
+          break loop;
+      }
+    }
+    return makeToken(PreprocessorParser.WS);
+  }
+
+  ProToken comment() throws IOException {
+    LOGGER.trace("Entering comment()");
+
+    // Escapes in comments are processed because you can end a comment
+    // with something dumb like: ~*~/
+    // We preserve that text.
+    // Note that macros are *not* expanded inside comments.
+    // (See the preprocessor source)
+    prepro.setDoingComment(true);
+    append(); // currChar=='*'
+    int commentLevel = 1;
+    while (commentLevel > 0) {
+      getChar();
+      unEscapedAppend();
+      if (currChar == '/') {
+        getChar();
+        unEscapedAppend();
+        if (currChar == '*')
+          commentLevel++;
+      } else if (currChar == '*') {
+        while (currChar == '*') {
+          getChar();
+          unEscapedAppend();
+          if (currChar == '/')
+            commentLevel--;
+        }
+      } else if (currInt == EOF_CHAR) {
+        prepro.lexicalThrow("Missing end of comment");
+      }
+    }
+    prepro.setDoingComment(false);
+    getChar();
+    return makeToken(PreprocessorParser.COMMENT);
+  }
+
+  ProToken singleLineComment() throws IOException {
+    LOGGER.trace("Entering singleLineComment()");
+
+    // Single line comments are treated just like regular comments,
+    // everything till end of line is considered comment - no escape
+    // character to look after
+    append(); // currChar=='/'
+
+    while (true) {
+      getChar();
+      unEscapedAppend();
+      if (currChar == '\r' || currChar == '\n' || currInt == EOF_CHAR) {
+        return makeToken(PreprocessorParser.COMMENT);
+      }
+    }
+  }
+
+  ProToken quotedString() throws IOException {
+    LOGGER.trace("Entering quotedString()");
+
+    // Inside quoted strings (string constants) we preserve
+    // the source code's original text - we don't discard
+    // escape characters.
+    // The preprocessor *does* expand macros inside strings.
+    for (;;) {
+      if (currInt == EOF_CHAR)
+        prepro.lexicalThrow("Unmatched quote");
+      unEscapedAppend();
+      if (currInt == currStringType && !prepro.isEscapeCurrent()) {
+        getChar();
+        if (currInt == currStringType) { // quoted quote
+          unEscapedAppend();
+        } else {
+          break; // close quote
+        }
+      }
+      getChar();
+    }
+
+    if (currChar == ':') {
+      boolean isStringAttributes = false;
+      // Preserve the colon before calling getChar,
+      // in case it belongs in the next token.
+      preserveCurrent();
+      String theText = ":";
+      for_loop : for (;;) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor the code to remove this label and the need for it. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3ALabelsShouldNotBeUsedCheck)